src/image/PhotoCalib.cc

```diff
@@ -293,6 +293,85 @@ MaskedImage<float> PhotoCalib::calibrateImage(MaskedImage<float> const &maskedIm
     return result;
 }
 
+afw::table::SourceCatalog PhotoCalib::calibrateCatalog(afw::table::SourceCatalog const &catalog,
+                                                       std::vector<std::string> const &instFluxFields) const {
+    auto const &inSchema = catalog.getSchema();
+    afw::table::SchemaMapper mapper(inSchema, true);  // true: share the alias map
+    mapper.addMinimalSchema(inSchema);
+
+    std::vector<table::Key<double>> instFluxKeys;
+    instFluxKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> instFluxErrKeys;
+    instFluxErrKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> fluxKeys;
+    fluxKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> fluxErrKeys;
+    fluxErrKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> magKeys;
+    magKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> magErrKeys;
+    magErrKeys.reserve(instFluxFields.size());
```

Comment:
This isn't bad by any means, but it strikes me as a place where a little local `struct` to hold those keys, might be a bit better - i.e. you'd just have a single `vector` of that struct, and then the inner `for` loop could just be a range based `for`, and there'd be a lot more simple attribute assignments instead of paren-filled `push_back()` and `back()` calls.

Reply:
Thanks for that suggestion. I think that does improve clarity.